nntrainer/models/neuralnet.cpp

```diff
@@ -739,16 +740,19 @@ int NeuralNetwork::train_run() {
     stat.num_iterations++;
   };
 
-  auto train_epoch_end = [this](RunStats &stat, DataBuffer &buffer) {
+  auto train_epoch_end = [this, stop_cb](RunStats &stat, DataBuffer &buffer) {
```

Comment:
Food for thought: from the on-device perspective, the number of epochs can be few (maybe even 1), and user might want to stop the training after a few iterations.
It might be worthwhile to design the implementation to later support stop_cb at iteration level as well.

Reply:
how about leaving this idea as todo and also adding the early_stopping  support.